### PR TITLE
DSS-5477: DatasetAuthorComponent

### PR DESCRIPTION
**Pull request by theseyi**
- Removes arbitrary `setTimeout`'s set for updating DOM elements when ready and uses Ember component lifecycle, converts component usage to inline, uses data-\* attribute in place of id reference to component element from JS
- Imports refactoring changes from linkedin branch into master

### CODE REVIEW

File: web/app/views/index.scala.html [hunk 1]
```diff
@@ -662,8 +662,7 @@
             {{/dataset-schema}}
           </div>
           <div id="ownertab" class="tab-pane">
-            {{#dataset-author owners=owners ownerTypes=ownerTypes showMsg=showMsg alertType=alertType ownerMessage=ownerMessage parentController=this}}
-            {{/dataset-author}}
+            {{dataset-author owners=owners ownerTypes=ownerTypes showMsg=showMsg alertType=alertType ownerMessage=ownerMessage parentController=this}}
```
Comment (luisah) [inline web/app/views/index.scala.html]: Suggestion, to improve readability you could have each attribute on its own line:

{{dataset-author 
    owners=owners 
    ownerTypes=ownerTypes 
    showMsg=showMsg 
    alertType=alertType 
    ownerMessage=ownerMessage 
    parentController=this}}

File: web/public/javascripts/components/components.js [hunk 2]
```diff
@@ -167,6 +167,30 @@ App.DatasetImpactComponent = Ember.Component.extend({
 });
 
 App.DatasetAuthorComponent = Ember.Component.extend({
+  didInsertElement() {
+    this._super(...arguments);
+    // Apply jQuery sortable plugin to element
+    this.$('[data-attribute=owner-table]')
```
Comment (igbopie) [inline web/public/javascripts/components/components.js]: Do you need to destroy sortable?

File: web/public/javascripts/routers/datasets.js [hunk 3]
```diff
@@ -201,41 +126,36 @@ App.DatasetsRoute = Ember.Route.extend({
...
-        datasetController.set("detailview", true);
-        if (params.originalSchema)
-        {
-          Ember.set(params, 'schema', params.originalSchema);
-        }
-        controller.set('model', params);
 
+    if (params && params.id) {
+      ({id, source, urn, name} = params);
+      let {originalSchema = null} = params;
```
Comment (igbopie) [inline web/public/javascripts/routers/datasets.js]: Should let be first here?

File: web/public/javascripts/routers/datasets.js [hunk 4]
```diff
@@ -546,65 +465,27 @@ App.DatasetRoute = Ember.Route.extend({
...
-                    var userEntitiesSource = Object.keys(userEntitiesMaps);
-                    controller.set("userEntitiesMaps", userEntitiesMaps);
-                    controller.set("userEntitiesSource", userEntitiesSource);
-                    setOwnerNameAutocomplete(controller);
-                  }
-                }
-              });
-            }, 500);
+    Promise.resolve($.get(`api/v1/datasets/${id}/owners`))
+        .then(({status, owners = []}) => (status === 'ok' && controller.set('owners', owners), void 0))
```
Comment (igbopie) [inline web/public/javascripts/routers/datasets.js]: Es6 is awesome but.... this is hard to understand hahaha!

Comment (theseyi) [inline web/public/javascripts/routers/datasets.js]: Haha, it's not ES6 but fair enough, I was being stingy and saving two lines
I'll update but it's equivalent to
```javascript
.then(({status, owners = []}) => {
  controller.set('owners', owners);
  return undefined; // not necessary with function braces (all funcs return by default) but 1:1 translation of above
})
...
`

**Pull request merged**